src/reanimated2/layoutReanimation/sharedTransitions/SharedTransition.ts

```diff
@@ -0,0 +1,331 @@
+  configureLayoutAnimations,
+  registerEventHandler,
+  unregisterEventHandler,
+} from '../../core';
+import { Platform } from 'react-native';
+import { runOnUIImmediately } from '../../threads';
+import './ProgressTransitionManager';
+
+const supportedProps = [
+  'width',
+  'height',
+  'originX',
+  'originY',
+  'transform',
+  'borderRadius',
+];
+
+type AnimationFactory = (
+  values: SharedTransitionAnimationsValues
+) => StyleProps;
+
+type TransitionProgressEvent = {
+  closing: number;
+  goingForward: number;
+  eventName: string;
+  progress: number;
+  target: number;
+};
+
+const dummyWorklet = () => {
+  'worklet';
+  // noop
+};
+
+export class SharedTransition {
+  private _customAnimationFactory: AnimationFactory | null = null;
+  private _animation: SharedTransitionAnimationsFunction | null = null;
+  private _transitionDuration = 500;
+  private _customProgressAnimation?: ProgressAnimation = undefined;
+  private _progressAnimation?: ProgressAnimation = undefined;
+  private _defaultTransitionType?: SharedTransitionType = undefined;
+  private static _sharedElementCount = 0;
+  private static _eventHandler = {
+    isRegistered: false,
+    onTransitionProgress: -1,
+    onAppear: -1,
+    onDisappear: -1,
+    onSwipeDismiss: -1,
+  };
+
+  public custom(customAnimationFactory: AnimationFactory): SharedTransition {
+    this._customAnimationFactory = customAnimationFactory;
+    return this;
+  }
+
+  public progressAnimation(
+    progressAnimationCallback: CustomProgressAnimation
+  ): SharedTransition {
+    this._customProgressAnimation = (viewTag, values, progress) => {
+      'worklet';
+      const newStyles = progressAnimationCallback(values, progress);
+      _notifyAboutProgress(viewTag, newStyles, true);
+    };
+    return this;
+  }
+
+  public duration(duration: number): SharedTransition {
+    this._transitionDuration = duration;
+    return this;
+  }
+
+  public defaultTransitionType(
+    transitionType: SharedTransitionType
+  ): SharedTransition {
+    this._defaultTransitionType = transitionType;
+    return this;
+  }
+
+  public getInstance(): SharedTransition {
+    return this;
+  }
+
+  public registerTransition(
+    viewTag: number,
+    sharedTransitionTag: string
+  ): void {
+    const transitionAnimation = this.getTransitionAnimation();
+    const progressAnimation = this.getProgressAnimation();
+    if (!this._defaultTransitionType) {
+      if (this._customAnimationFactory && !this._customProgressAnimation) {
+        this._defaultTransitionType = SharedTransitionType.ANIMATION;
+      } else {
+        this._defaultTransitionType = SharedTransitionType.PROGRESS_ANIMATION;
+      }
+    }
+    configureLayoutAnimations(
+      viewTag,
+      LayoutAnimationType.SHARED_ELEMENT_TRANSITION,
+      transitionAnimation,
+      sharedTransitionTag
+    );
+    configureLayoutAnimations(
+      viewTag,
+      LayoutAnimationType.SHARED_ELEMENT_TRANSITION_PROGRESS,
+      dummyWorklet,
```

Comment:
Can you explain this in a comment? Why there is an empty method added as layout animation?

Reply:
Resolved in: https://github.com/software-mansion/react-native-reanimated/pull/4421/commits/5ab12a598e5f2b64cfa2441e68cdbc4bffb7293f